crates/gpui/src/platform/linux/platform.rs

```diff
@@ -481,7 +481,11 @@ impl<P: LinuxClient + 'static> Platform for P {
                     let username = attributes
                         .get("username")
                         .ok_or_else(|| anyhow!("Cannot find username in stored credentials"))?;
-                    let secret = item.secret().await?;
+                    // oo7 panics if the retrieved secret can't be decrypted due to
+                    // unexpected padding.
+                    let secret = std::panic::catch_unwind(|| item.secret())
```

Comment:
Doesn't secret `await` before trying to parse the secret? I think this catch_unwind won't catch this panic :)

Reply:
Good point, I was thinking it had to be the call to `Item::secret` itself because it's on the stack when the panic happens, but given that it's an async fn I guess that just means the panic happens during the first poll.